Calendar/Calendar.js

```diff
@@ -184,35 +171,38 @@ class Calendar extends PureComponent<Props, State> {
     );
   }
 
-  renderWeeks = (date, monthIndex) => (
-    <View key={date} style={styles.calRow}>
-      {[0, 1, 2, 3, 4, 5, 6].map((i) => {
-        const daysDate = date.clone().add(i, 'd');
-        const isCurrentMonth = daysDate.month() === monthIndex;
-        const isPast = daysDate.diff(moment(), 'd') < 0;
-
-        const extraStyle = !isPast && isCurrentMonth ? { color: 'black', fontWeight: 'bold' } : { color: 'grey' };
-
-        const isSelected = this.props.selected && moment(this.props.selected).diff(daysDate, 'd') === 0;
-
-        return (
-          <TouchableOpacity
-            key={`${daysDate}${i}`}
-            style={[
-              styles.calCell,
-              this.dayCellDimension,
-              isSelected ? styles.isSelectedContainer : {},
-            ]}
-            onPress={() => !isPast && this.handleCellPress(daysDate)}
-          >
-            <Text style={[styles.calText, extraStyle, isSelected ? styles.isSelectedText : {}]}>
-              {daysDate.format('D')}
-            </Text>
-          </TouchableOpacity>
-        );
-      })}
-    </View>
-  );
+  renderWeeks = (date, monthIndex) => {
+    const { selected } = this.props;
+    return (
+      <View key={date} style={styles.calRow}>
+        {[0, 1, 2, 3, 4, 5, 6].map((i) => {
+          const daysDate = date.clone().add(i, 'd');
+          const isCurrentMonth = daysDate.month() === monthIndex;
+          const isPast = daysDate.diff(moment(), 'd') < 0;
+
+          const extraStyle = !isPast && isCurrentMonth ? { color: 'black', fontWeight: 'bold' } : { color: 'grey' };
+
+          const isSelected = selected && moment(selected).diff(daysDate, 'd') === 0;
+
+          return (
+            <TouchableOpacity
+              key={`${daysDate}${i}`}
+              style={[
+                styles.calCell,
+                this.dayCellDimension,
+                isSelected ? styles.isSelectedContainer : {},
+              ]}
+              onPress={() => !isPast && this.handleCellPress(daysDate)}
+            >
+              <Text style={[styles.calText, extraStyle, isSelected ? styles.isSelectedText : {}]}>
```

Comment:
make a function to return a style object with param isPast, isCurrentMont and isSelected
```javascript
function getSomeStyleFunction(isPast, isCurMonth, isSelected) {
 
   if(!isPast && isCurrentMonth && isSelected) {
         
      return {
         color: 'black', 
         fontWeight: 'bold',
         borderRadius: 5,
         backgroundColor: BRAND_COLOR
    }
  return { color: 'grey' }
}
}
```
```javascript
  <Text style={[styles.calText, getSomeStyleFunction(isPast, isCurMonth, isSelected) ]}>
```